testool/src/statetest/parse.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::HashMap;
+
+use crate::abi;
+use crate::Compiler;
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use eth_types::evm_types::OpcodeId;
+use eth_types::{Address, Bytes, H256, U256};
+use log::debug;
+
+type Label = String;
+
+/// returns the element as an address
+pub fn parse_address(as_str: &str) -> Result<Address> {
+    if let Some(hex) = as_str.strip_prefix("0x") {
+        Ok(Address::from_slice(
+            &hex::decode(hex).context("parse_address")?,
+        ))
+    } else {
+        Ok(Address::from_slice(
+            &hex::decode(as_str).context("parse_address")?,
+        ))
+    }
```

Comment:
Nitpick: I see many functions that use `strip_prefix` have duplicated lines. We could deduplicate the expression further. It's fine if we don't do it now.
```suggestion
    let hex = as_str.strip_prefix("0x").unwrap_or(as_str);
    Ok(Address::from_slice(
        &hex::decode(hex).context("parse_address")?,
    ))
```